sdk/core/azure-core/src/test/java/com/azure/core/http/rest/RestProxyTests.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import com.azure.core.exception.UnexpectedLengthException;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.util.FluxUtil;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.nio.ByteBuffer;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+/**
+ * Tests {@link RestProxy}.
+ */
+public class RestProxyTests {
+    private static final byte[] EXPECTED = new byte[] { 0, 1, 2, 3, 4 };
+
+    @Test
+    public void emptyRequestBody() {
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET, "http://localhost");
+
+        StepVerifier.create(RestProxy.validateLength(httpRequest))
+            .verifyComplete();
+    }
+
+    @Test
+    public void expectedBodyLength() {
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET, "http://localhost")
+            .setBody(EXPECTED)
+            .setHeader("Content-Length", "5");
+
+        StepVerifier.create(collectRequest(httpRequest))
+            .assertNext(bytes -> assertArrayEquals(EXPECTED, bytes))
+            .verifyComplete();
+    }
+
+    @Test
+    public void unexpectedBodyLength() {
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET, "http://localhost")
+            .setBody(EXPECTED);
+
+        StepVerifier.create(collectRequest(httpRequest.setHeader("Content-Length", "4")))
+            .verifyError(UnexpectedLengthException.class);
+
+        StepVerifier.create(collectRequest(httpRequest.setHeader("Content-Length", "6")))
+            .verifyError(UnexpectedLengthException.class);
```

Comment:
Would be better to validate the error message as well as that is what we will use to debug issues.

```java
        StepVerifier.create(collectRequest(httpRequest.setHeader("Content-Length", "4")))
            .verifyErrorSatisfies(throwable -> {
                    assertTrue(throwable instanceof UnexpectedLengthException);
                    assertTrue(throwable.getMessage().equals("Request body emitted 4 bytes, less than the expected 5 bytes."
            });

        StepVerifier.create(collectRequest(httpRequest.setHeader("Content-Length", "6")))
            .verifyErrorSatisfies(throwable -> {
                    assertTrue(throwable instanceof UnexpectedLengthException);
                    assertTrue(throwable.getMessage().equals("Request body emitted 6 bytes, more than the expected 5 bytes."
            });
```

Reply:
Added